app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala

```diff
@@ -113,7 +113,7 @@ object NSIPayload {
       BankDetails(
         bankDetails.sortCode.cleanupSpecialCharacters.removeAllSpaces.filterNot(allowedSortCodeSeparators.contains).grouped(2).mkString("-"),
         bankDetails.accountNumber.cleanupSpecialCharacters.removeAllSpaces,
-        bankDetails.rollNumber.map(_.cleanupSpecialCharacters.removeAllSpaces),
+        bankDetails.rollNumber,
```

Comment:
I think we'd still want to clean up special characters. That does this:
```scala
def cleanupSpecialCharacters: String = s.replaceAll("\t|\n|\r", " ").trim.replaceAll("\\s{2,}", " ")
```
which trims tabs, new lines, carriage returns and trims trailing and leading spaces

Reply:
the frontend already takes care of those as it doesn't allow the spaces and `\t|\n|\r` ?

Reply:
that's true - but the API service won't have that logic in place

Reply:
yeah, i added a comment to the other PR if we need to add the validation in the api layer as well ?
is there any reason why we are not validating the bank details in the api layer ?

Reply:
for the API journey there's no real incentive to have the checks up front as we'd get the same result if the request made it's way to the proxy. In the FE the reason we have the logic there as well is that there we can stop people on the journey if their bank details are wrong before they create an account, but we can't do that sort of thing in the API journey

Reply:
ok, i added the logic back